### PR TITLE
sci-physics/geant-vmc: Version bump and fixes

### PR DESCRIPTION
**Pull request by olifre**
This series contains:
- EAPI bump of existing versions. 
- Adapt to the `c++11` USE-flag of `sci-physics/root`.
- Block `>sci-physics/geant-4.10.05` for in-tree `=sci-physics/geant-vmc-4.3.6` (fails to compile against this version). 
- Add `=sci-physics/geant-vmc-4.4.0` which is compatible with most recent `sci-physics/geant` again.

### CODE REVIEW

File: sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Copyright 1999-2018 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
```
Comment (mgorny) [inline sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild]: This is the kind of change that really desires a revbump (`ebump -c` is your friend).

File: sci-physics/geant-vmc/geant-vmc-4.4.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+IUSE="doc examples geant3 +g4root +mtroot vgm test"
+
+# sci-physics/root[c++11] required to match sci-physics/geant flags.
+RDEPEND="
+	>=sci-physics/geant-4.10.03:=[opengl,geant3?]
+	sci-physics/root:=[c++11]
+	vgm? ( >=sci-physics/vgm-4.4:= )"
+DEPEND="${RDEPEND}
+	doc? ( app-doc/doxygen )"
+RESTRICT="
```
Comment (mgorny) [inline sci-physics/geant-vmc/geant-vmc-4.4.0.ebuild]: `!test? ( test )` too

File: sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild (hunk 1 above)
Comment (olifre) [reply sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild]: Thanks, also for the hint to `ebump` which I did not know about yet!

File: sci-physics/geant-vmc/geant-vmc-4.4.0.ebuild (hunk 2 above)
Comment (olifre) [reply sci-physics/geant-vmc/geant-vmc-4.4.0.ebuild]: Fixed now.

File: sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild [hunk 3]
```diff
@@ -23,18 +23,19 @@ LICENSE="GPL-2"
 SLOT="4"
 IUSE="doc examples geant3 +g4root +mtroot vgm test"
 
-# sci-physics/root[root7] flag activates std=c++14, which we could only support if also Geant is built with it.
```
Comment (mgorny) [inline sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild]: Since you're fixing this in revbump, there's no need to edit the old version.

Comment (olifre) [reply sci-physics/geant-vmc/geant-vmc-4.3.6.ebuild]: Would you prefer if I revert this change, or leave as-is (since you already approved the changes)?

**Comment by olifre** [comment]
> after spending 4 hours building ROOT and dependencies just to discover it fails at the very end due to build system bug

Thanks! Remember that this (sadly) also requires `sci-physics/geant` which is both large and very compile-time intensive.

**Comment by olifre** [comment]
I had to push one more commit to fix test suite issues. I made three mistakes which are now adressed:
- I was missing `!examples? ( test )` in `RESTRICT` - examples are used by the tests. 
- I had garfield installed (not packaged yet), which the test suite automagically uses. It fails if it is not there. The flag to turn garfield-tests off is already included in upstream's master, I patched this via `sed` in the ebuild now. 
- Two tests actually failed on a vanilla Gentoo machine. This is now adressed in https://github.com/vmc-project/geant4_vmc/pull/11 and I "backported" the patch by another `sed` in the ebuild. 

All tests run fine now for me on a system without Garfield installed. I also made use of the `garfield=off` flags for the `9999` ebuild.

**Comment by olifre** [comment]
> Fails to build for me:

Many thanks for testing with an uncommon USE flag combination on `sci-physics/root`, I did not catch this dependency up to now! 
I have pushed an update changing the following:
- Depend on `sci-physics/root[vmc]` (this should fix your build error). 
- Leave the non-revbumped ebuild (`=sci-physics/geant-vmc-4.3.6`) untouched.

**Comment by mgorny** [comment]
Uncommon? I was trying to save time by building as few as possible! ;-P

**Comment by olifre** [comment]
> Uncommon? I was trying to save time by building as few as possible! ;-P

I'm glad you did, I never encountered a `sci-physics/root` without `vmc` :wink: . Sadly, you will not be able to save much time whatever you try - the embedded clang in ROOT eats most of the compile time (and it can't easily be taken out since there are patches on top of it which can't be upstreamed easily ( https://sft.its.cern.ch/jira/browse/ROOT-8605 ).

**Comment by mgorny** [comment]
I suppose we're almost done here:

```
 * Messages for package sci-physics/geant-vmc-4.4.0:
 * Log file: /var/log/portage/sci-physics:geant-vmc-4.4.0:20190718-045221.log

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/g4root_OpNovice
```

See also https://wiki.gentoo.org/wiki/Package_testing#make.conf for the knobs to detect this stuff.

**Comment by olifre** [comment]
@mgorny That's strange - I have those flags in my `make.conf` and do not observe this warning, and I also see the `-frecord-gcc-switches` call in the build log when `g4root_OpNovice` is compiled. 

Could you please share your build log? 
My guess is that `sci-physics/geant-vmc` is adopting `CFLAGS` from `sci-physics/geant` (which I also compiled with the test-flags) which would hide the issue for me. If that's the case, I need to delve into upstream's CMake to fix it.

**Comment by mgorny** [comment]
[sci-physics:geant-vmc-4.4.0:20190718-045221.log](https://github.com/gentoo/gentoo/files/3406744/sci-physics.geant-vmc-4.4.0.20190718-045221.log)

Actually, it seems to be taking flags from ROOT (I had to change them because of clang use).

**Comment by mgorny** [comment]
I guess it's non-trivial then. Let me know if you want to fix it or if I should merge as-is.

**Comment by olifre** [comment]
I have a hunch where it goes wrong (there is one upstream change not propagated to this build target) - I'll try a patch and if it works, will let you know in the next half hour (and also submit this to upstream) :wink:.

**Comment by olifre** [comment]
Done - that patch seems to fix it (the issue comes from a CFLAGS overriding bug inside a Geant-provided file). Can you confirm?

**Pull request closed without merging**